app/src/main/java/com/gholem/moneylab/features/planning/adapter/PlanningAdapter.kt

```diff
@@ -0,0 +1,66 @@
+package com.gholem.moneylab.features.planning.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.gholem.moneylab.R
+import com.gholem.moneylab.databinding.ItemPlanningBinding
+import com.gholem.moneylab.domain.model.TransactionModel
+import com.gholem.moneylab.features.planning.adapter.viewholder.PlanningViewHolder
+import timber.log.Timber.i
+
+class PlanningAdapter(val positionClickListener: (position: Int) -> Unit) :
+    RecyclerView.Adapter<PlanningViewHolder>() {
+
+    private var adapterData: List<TransactionModel> = emptyList()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlanningViewHolder {
+        return when (viewType) {
```

Comment:
If you have only one view type then just trigger here `creativeViewHolder(parent)` method without any checks, it's unnecessary.